oauth2client/transport/__init__.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TODO"""
+
+from six.moves import http_client
+
+from . import httplib2
+
+# Google Data client libraries may need to set this to [401, 403].
+REFRESH_STATUS_CODES = (http_client.UNAUTHORIZED,)
+
+_default_transport = httplib2
+
+
+def set_default_transport(transport_module):
+    """TODO"""
+    global _default_transport
+    _default_transport = transport_module
+
+
+def get_default_transport():
+    """TODO"""
+    return _default_transport
+
+
+def get_cached_http():
+    """TO BE REMOVED"""
+    return _default_transport.get_cached_http()
+
+
+def get_http_object(*args, **kwargs):
+    """TODO"""
+    return _default_transport.get_http_object(*args, **kwargs)
+
+
+def inject_credentials(http_object, credentials):
+    """TODO"""
+    return _default_transport.inject_credentials(
+        http_object, credentials, refresh_status_codes=REFRESH_STATUS_CODES)
+
+
+def inject_assertion_credentials(http_object, credentials):
+    """TODO"""
+    # TODO: Consolidate with inject_credentials.
+    return _default_transport.inject_assertion_credentials(
+        http_object, credentials, refresh_status_codes=REFRESH_STATUS_CODES)
+
+
+def request(http_object, uri, method='GET', body=None, headers=None, **kwargs):
```

Comment:
Once I have a solidified interface this will be the function that declares it, so I think it's important to have some fixed args that all transports need to accept.